### PR TITLE
Drop pglogical subscriptions before restore

### PR DESCRIPTION
**Pull request by carbonin**
pglogical subscriptions cause a postgresql worker process to hold open connections to the database. If we attempt to drop the database while we still have subscriptions, we will always fail.

This commit removes any existing subscriptions before attempting a restore and ignores the failures in the case that pglogical is not being used. This allows the database restore to complete cleanly.

https://bugzilla.redhat.com/show_bug.cgi?id=1393049

@Fryguy @gtanzillo please review

### CODE REVIEW

File: lib/gems/pending/appliance_console.rb [hunk 1]
```diff
@@ -393,6 +391,8 @@ module ApplianceConsole
             File.delete(DB_RESTORE_FILE)
           elsif !rake_success
             say("\nDatabase restore failed")
+            connections = ApplianceConsole::Utilities.db_connections - 1
```
Comment (gtanzillo) [inline lib/gems/pending/appliance_console.rb]: Does this always assume that there is a an active pglogical connection?

Comment (carbonin) [inline lib/gems/pending/appliance_console.rb]: No, the `db_connections` method actually connects to the database. So the `- 1` is for the one we are currently using to query.

**Comment by chessbyte** [comment]
Euwe Backport conflict:
```sh
$ git cherry-pick -e -x -m 1 -Xsubtree=../manageiq-gems-pending 098cab8
error: could not apply 098cab8... Merge pull request #23 from carbonin/clear_pglogical_subscriptions_before_restore
hint: after resolving the conflicts, mark the corrected paths
hint: with 'git add <paths>' or 'git rm <paths>'
hint: and commit the result with 'git commit'

$ git status
On branch euwe
Your branch is up-to-date with 'upstream/euwe'.
You are currently cherry-picking commit 098cab8.
  (fix conflicts and run "git cherry-pick --continue")
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

Changes to be committed:

	modified:   gems/pending/spec/util/postgres_admin_spec.rb
	modified:   gems/pending/util/postgres_admin.rb

Unmerged paths:
  (use "git add/rm <file>..." as appropriate to mark resolution)

	deleted by us:   lib/gems/pending/appliance_console.rb

$ git diff
* Unmerged path lib/gems/pending/appliance_console.rb
```

**Comment by chessbyte** [comment]
@carbonin @gtanzillo Please create Euwe-specific PR after resolving conflict or suggest ways I can do that locally.

**Comment by chessbyte** [comment]
Backported to Euwe via https://github.com/ManageIQ/manageiq/pull/12900

**Pull request merged**